internal/pkg/instrumentation/manager.go

```diff
@@ -128,7 +147,10 @@ func (m *Manager) registerProbe(p probe.Probe) error {
 		return err
 	}
 
-	m.probes[id] = p
+	m.probes[id] = ProbeReference{
+		probe:   p,
+		closers: make([]io.Closer, 0),
```

Comment:
This is not needed. It is only allocating memory prior to need. `append` handles appends to empty values without issue.

```suggestion
```